torch/distributed/distributed_c10d.py

```diff
@@ -434,6 +434,7 @@ def __init__(self, op: Callable, tensor: torch.Tensor, dst_tensor: Optional[torc
 # Use them through the _world object to make sure the _world override mechanism
 _pg_map: Dict[ProcessGroup, Tuple[str, Store]] = {}
 _pg_names: Dict[ProcessGroup, str] = {}
+_pg_descs: Dict[ProcessGroup, str] = {}
```

Comment:
Hmm do we need this? Since we can get the descriptive name via `pg.pg_descs`, is there a reason that we want to get it via  `_pg_descs[pg]`?

IIUC these dicts here are for adding python-only properties to pgs without having to add them in C++ (I think `_pg_names` here is not needed anymore).

Reply:
i dont know the history but if we can just make `_get_process_group_desc` return `pg.pg_descs` it seems better to not add all the extra python code.

Reply:
Thanks for pointing this out. Yea it shouldn't be necessary to have a separate dict for pg_descs. For pg_names, I think the main purpose to help check if the group_name is unique.

Reply:
addressed comments